modules/ensemble/lib/framework/device.dart

```diff
@@ -13,22 +14,95 @@ import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
 import 'package:get_it/get_it.dart';
 import 'package:intl/intl.dart';
 
-/// get device information as well as requesting device permissions
-class Device
-    with
-        Invokable,
-        MediaQueryCapability,
-        LocationCapability,
-        DeviceInfoCapability {
+class Device with Invokable, MediaQueryCapability, LocationCapability, DeviceInfoCapability, WidgetsBindingObserver {
   static final Device _instance = Device._internal();
+  static bool _isInitialized = false;
+  bool _isHandlingChange = false;
+  MediaQueryData? _lastReportedData;
+
+  static final StreamController<MediaQueryData> _deviceUpdateController = 
+      StreamController<MediaQueryData>.broadcast();
+  static Stream<MediaQueryData> get onDeviceUpdate => _deviceUpdateController.stream;
+  
+  Device._internal() {
+    if (!_isInitialized) {
```

Comment:
Device is a singleton, meaning there's only 1 instance of it. You don't need to check if it's initialized or not.